src/main/java/org/broadinstitute/consent/http/resources/LibraryCardResource.java

```diff
@@ -0,0 +1,118 @@
+
+  @Inject
+  public LibraryCardResource(
+    UserService userService,
+    LibraryCardService libraryCardService
+  ) {
+    this.userService = userService;
+    this.libraryCardService = libraryCardService;
+  }
+
+  @GET
+  @Produces("application/json")
+  @RolesAllowed(ADMIN)
+  public Response getLibraryCards(@Auth AuthUser authUser) {
+    try{
+      List<LibraryCard> libraryCards = libraryCardService.findAllLibraryCards();
+      return Response.ok().entity(libraryCards).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @GET
+  @Produces("application/json")
+  @Path("/{id}")
+  @RolesAllowed(ADMIN)
+  public Response getLibraryCardById(@Auth AuthUser authUser, @PathParam("id") Integer id) {
+    try {
+      LibraryCard libraryCard = libraryCardService.findLibraryCardById(id);
+      return Response.ok().entity(libraryCard).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @GET
+  @Produces("application/json")
+  @Path("/institution/{id}")
+  @RolesAllowed(ADMIN)
+  public Response getLibraryCardsByInstitutionId(@Auth AuthUser authUser, @PathParam("id") Integer id) {
+    try{
+      List<LibraryCard> libraryCards = libraryCardService.findLibraryCardsByInstitutionId(id);
+      return Response.ok().entity(libraryCards).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @POST
+  @Consumes("application/json")
+  @Produces("application/json")
+  @RolesAllowed(ADMIN)
+  public Response createLibraryCard(@Auth AuthUser authUser, String libraryCard) {
+    try{
+      User user = userService.findUserByEmail(authUser.getName());
+      LibraryCard payload = new Gson().fromJson(libraryCard, LibraryCard.class);
+      LibraryCard newLibraryCard = libraryCardService.createLibraryCard(payload, user.getDacUserId());
+      return Response.ok().entity(newLibraryCard).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @PUT
+  @Consumes("application/json")
+  @Produces("application/json")
+  @Path("/{id}")
+  @RolesAllowed(ADMIN)
+  public Response updateLibraryCard(@Auth AuthUser authUser, @PathParam("id") Integer id, String libraryCard) {
+    try {
+      User user = userService.findUserByEmail(authUser.getName());
+      LibraryCard payload = new Gson().fromJson(libraryCard, LibraryCard.class);
+      LibraryCard updatedLibraryCard = libraryCardService.updateLibraryCard(payload, id, user.getDacUserId());
+      return Response.ok().entity(updatedLibraryCard).build();
+    } catch(Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @DELETE
+  @Produces("application/json")
+  @Path("/{id")
+  @RolesAllowed(ADMIN)
+  public Response deleteLibraryCard(@Auth AuthUser authUser, @PathParam("id") Integer id) {
+    try {
+      libraryCardService.deleteLibraryCardById(id);
+      return Response.status(204).build();
```

Comment:
In almost all of our other DELETE endpoints, we return a 200. The only exception is the institution delete, which I must not have looked at closely the first time around. I needed to go back to my [history books](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE) to sort out the right responses here: 

> If a DELETE method is successfully applied, there are several response status codes possible:
>  
> A 202 (Accepted) status code if the action will likely succeed but has not yet been enacted.
> A 204 (No Content) status code if the action has been enacted and no further information is to be supplied.
> A 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.

So this is correct. I think we need another tech-debt ticket to address how we're handling `DELETE`'s and align their response codes appropriately.